### ISSUE TITLE
"FuncDiv" for contirbutional/functional diversity (alpha and beta diversities)?

### ISSUE THREAD

**Issue by marwa38** [ChiLiubio/microeco#200]
Hi team 
This is a powerful package and your up-to-date tutorial is very helpful. 
I was wondering if you could help with the downstream analysis for functional diversity?  https://github.com/gavinmdouglas/FuncDiv
`FuncDiv` is a new package developed by the picrsut2 developers for different workflows and not restricted to microbiota. 
I was wondering if you could help by giving just an example tutorial for it.
Running the codes for functional alpha and beta diversity is straightforward but the downstream analysis needs some inputs from the microbiome world. 
In the meanwhile if you please could suggest some ideas for downstream analysis? that would be great. 
Cheers
Marwa

**Comment by ChiLiubio** [comment]
Thanks. I will try to figure it out. Do you mean using the microeco package to do "downstream analysis" of the output of FuncDiv package?

Chi

**Comment by marwa38** [comment]
> "downstream analysis" of the output of FuncDiv package?

yes, the packaging paper is straightforward and short and intentionally left without option what are the options for downstream analysis regarding the microbiome world? 
Here is more info: https://twitter.com/gavin_m_douglas/status/1622951069293903874

**Comment by marwa38** [comment]
For the functional analysis, it would great to help with the downstream analysis of `POMS `this was developed by one of the main contributors to picrust2 but again needs help with downstream analysis. https://twitter.com/gavin_m_douglas/status/1595072133071347712
https://academic.oup.com/bioinformatics/article/38/22/5055/6731923?login=true
https://github.com/gavinmdouglas/POMS

**Comment by ChiLiubio** [comment]
Thanks very much. I will learn more on it.

**Comment by marwa38** [comment]
[ggpicrust2](https://github.com/cafferychen777/ggpicrust2) a new package for visualisation of picrust2 especially that STAMP haven't been updated since 2015 that might be good

**Comment by ChiLiubio** [comment]
Yes. I also find it. Thanks.

**Comment by ChiLiubio** [comment]
Hi. Sorry on that It was put on hold. Maybe the primary reason is I have no idea about how to extend functional diversity analysis based on the current framework. Please feel free to tell me if you have more detailed need on the analysis.

**Comment by marwa38** [comment]
Some suggestions are below that would be nice to consider:
I think also what are the specific functions that need to have RDA?
![image](https://github.com/ChiLiubio/microeco/assets/68851733/6d6f1415-047c-468c-91c2-ae7419713792)
I think you are already have those functions active in microeco but will divert to the functional/contributional diversity:
![image](https://github.com/ChiLiubio/microeco/assets/68851733/677593bd-591e-49ce-b18f-0b4eb2dc9eee)
![image](https://github.com/ChiLiubio/microeco/assets/68851733/2aaba91f-f67f-405f-92c8-e3fdf88d38cd)
How you find that?

**Comment by marwa38** [comment]
@ChiLiubio 

Just checked this paper referring to functional diversity with a simple figure https://www.nature.com/articles/s41398-023-02325-5/figures/3

**Comment by ChiLiubio** [comment]
Hi
These analysis methods can be easily achieved by the microeco package. The primary thing is to convert functional data to microtable object like what we show in the functional prediction part (https://chiliubio.github.io/microeco_tutorial/explainable-class.html#trans_func-class). Then all the operations are very similar with the steps in the tutorial. One notable thing is when you use cal_alphadiv function, please add parameter `measures = "Shannon"`.
